src/creatures/combat/condition.cpp

```diff
@@ -1460,6 +1496,276 @@ void ConditionDamage::generateDamageList(int32_t amount, int32_t start, std::lis
+	Position creaturePos = creature->getPosition();
+
+	int_fast32_t dx = Position::getDistanceX(creaturePos, fleeingFromPos);
+	int_fast32_t dy = Position::getDistanceY(creaturePos, fleeingFromPos);
+
+	int_fast32_t offx = Position::getOffsetX(creaturePos, fleeingFromPos);
+	int_fast32_t offy = Position::getOffsetY(creaturePos, fleeingFromPos);
+
+	// Discover where the monster is
+	if (offx == 0 && offy == 0) {
+		/*
+		*	Monster is on the same SQM of the player
+		*	Flee to Anywhere
+		*/
+		SPDLOG_DEBUG("[ConditionsFeared::getFleeDirection] Monster is on top of player, flee randomly. {} {}", offx, offy);
+		return getRandomDirection(creature, creaturePos);
+	} else if (offx >= 1 && offy <= 0 ) {
+		/*
+		*	Monster is on SW Region
+		*	Flee to N(0), NE(1) and E(2)
+		*/
+		SPDLOG_DEBUG("[ConditionsFeared::getFleeDirection] Monster is on the SW region, flee to N, NE or E. {} {}", offx, offy);
+		
+		if (offy == 0) {
+			this->fleeIndx = 2; // Starts at East
+		} else {
+			this->fleeIndx = 0; // Starts at North
+		}
+		
+		return true;
+	} else if (offx >= 0 && offy >= 1 ) {
+		/*
+		*	Monster is on NW Region
+		*	Flee to E(2), SE(3) and S(4)
+		*/
+		SPDLOG_DEBUG("[ConditionsFeared::getFleeDirection] Monster is on the NW region, flee to E, SE or S. {} {}", offx, offy);
+		
+		if (offx == 0) {
+			this->fleeIndx = 4; // Starts at South
+		} else {
+			this->fleeIndx = 2; // Starts at East
+		}
+		
+		return true;
+	} else if (offx <= -1 && offy >= 0 ) {
+		/*
+		*	Monster is on NE Region
+		*	Flee to S(4), SW(5) and W(6)
+		*/
+		SPDLOG_DEBUG("[ConditionsFeared::getFleeDirection] Monster is on the NE region, flee to S, SW or W. {} {}", offx, offy);
+		
+		if (offy == 0) {
+			this->fleeIndx = 6; // Starts at West
+		} else {
+			this->fleeIndx = 4; // Starts at South
+		}
+
+		return true;
+	} else if (offx <= 0 && offy <= -1 ) {
+		/*
+		*	Monster is on SE
+		*	Flee to W(6), NW(7) and N(0)
+		*/
+		SPDLOG_DEBUG("[ConditionsFeared::getFleeDirection] Monster is on the SE region, flee to W, NW or N. {} {}", offx, offy);
+		
+		if (offx == 0) {
+			this->fleeIndx = 0; // Starts at North
+		} else {
+			this->fleeIndx = 6; // Starts at West
+		}
+		
+		return true;
+	}
+
+	SPDLOG_DEBUG("[ConditionsFeared::getFleeDirection] Something went wrong. {} {}", offx, offy);
+	return false;
+}
+
+bool ConditionFeared::getFleePath(Creature* creature, Position& pos, std::forward_list<Direction>& dirList) {
+	const std::vector<uint8_t> walkSize { 15, 9, 3, 1 };
+	bool found = false;
+	uint8_t found_size = 0;
+	Position futurePos = pos; //getNextPosition(fleeDir, currentPos);
+
+	do {
+		for (uint8_t wsize : walkSize) {
+			SPDLOG_DEBUG("[ConditionsFeared::getFleePath] Checking on index {} with walkSize of {}", fleeIndx, wsize);
+
+			if(fleeIndx == 8) { // Reset index if at the end of the loop
+				fleeIndx = 0;
+			}
+
+			if(!isStuck(creature, pos)){ // Check if it is possible to walk to any direction
```

Comment:
after fixing the `isStuck` logic, this if condition must not be negated
```suggestion
			if(isStuck(creature, pos)){ // Check if it is possible to walk to any direction
```